src/components/CheckoutGrid/Table.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useMemo } from 'react'
+import { useSelector } from 'react-redux'
+import { getCart } from '../../store/cartSlice'
+import { CheckoutGridTable } from './styles'
+
+export default function Table () {
+  const { products } = useSelector(getCart)
+
+  const calculateTotal = useMemo(() => {
```

Comment:
I see this function in the `CartTable` component, so for these scenarios where you need to calculate the total amount in more than one component is more convenient to put this logic in the global state.